sequence_processing_pipeline/BCL2FASTQJob.py

```diff
@@ -0,0 +1,166 @@
+
+    def _validate_bcl_directory(self):
+        bcl_directory = join(self.root_dir, 'Data', 'Intensities', 'BaseCalls')
+        bcl_count = 0
+        if exists(bcl_directory):
+            for root, dirs, files in walk(bcl_directory):
+                for some_file in files:
+                    # subdirectories and files other than '.bcl' files from
+                    # bcl_directory are to be expected.
+                    if some_file.endswith('.bcl') or some_file.endswith('.cbcl'):
+                        bcl_count += 1
+        else:
+            logging.debug("input_directory '%s' does not contain subdirectory Data/Intensities/BaseCalls." % self.root_dir)
+            raise PipelineError("input_directory '%s' does not contain subdirectory Data/Intensities/BaseCalls." % self.root_dir)
+
+        if bcl_count < 1:
+            # we can increase to whatever threshold is acceptable.
+            raise PipelineError("input_directory '%s' does not contain enough bcl files (%d)." % (bcl_directory, bcl_count))
+
+    def _generate_job_script(self, queue_name, node_count, nprocs, wall_time_limit):
+        # TODO: Use Jinja template instead
+        lines = []
+
+        lines.append("#!/bin/bash")
+        # The Torque equiv for calling SBATCH on this script and supplying params
+        # w/environment variables is to do the same on QSUB but with -v instead of
+        # --export. The syntax of the values are otherwise the same.
+        # -v <variable[=value][,variable2=value2[,...]]>
+
+        # declare a name for this job to be sample_job
+        lines.append("#PBS -N %s" % self.job_name)
+
+        # what torque calls a queue, slurm calls a partition
+        # SBATCH -p SOMETHING -> PBS -q SOMETHING
+        # (Torque doesn't appear to have a quality of service (-q) option. so this
+        # will go unused in translation.)
+        lines.append("#PBS -q %s" % queue_name)
+
+        # request one node
+        # Slurm --ntasks-per-node=<count> -> -l ppn=<count>	in Torque
+        lines.append("#PBS -l nodes=%d:ppn=%d" % (node_count, nprocs))
+
+        # Slurm --export=ALL -> Torque's -V
+        lines.append("#PBS -V")
+
+        # Slurm walltime limit --time=24:00:00 -> Torque's -l walltime=<hh:mm:ss>
+        # using the larger value found in the two scripts (36 vs 24 hours)
+        lines.append("#PBS -l walltime=%d:00:00" % wall_time_limit)
+
+        # send email to charlie when a job starts and when it terminates or
+        # aborts. This is used to confirm the package's own reporting
+        # mechanism is reporting correctly.
+        lines.append("#PBS -m bea")
+
+        # specify your email address
+        # TODO: Send an email to jeff, and qiita.help as well.
+        lines.append("#PBS -M ccowart@ucsd.edu")
+
+        # min mem per CPU: --mem-per-cpu=<memory> -> -l pmem=<limit>
+        # taking the larger of both values (10G > 6G)
+        #lines.append("#PBS -l pmem=10gb")
```

Comment:
readd this line and use what's expected; also set that value as a global -> to change to conf